shared/fs/destination-picker/container.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import {compose, connect, setDisplayName} from '../../util/container'
+import memoize from 'memoize-one'
+import DestinationPicker from '.'
+import * as Types from '../../constants/types/fs'
+import * as Constants from '../../constants/fs'
+import * as FsGen from '../../actions/fs-gen'
+import {navigateUp} from '../../actions/route-tree'
+
+const mapStateToProps = state => ({
+  _moveOrCopy: state.fs.moveOrCopy,
+  _pathItems: state.fs.pathItems,
+})
+
+const mapDispatchToProps = dispatch => ({
+  onCancel: () => dispatch(navigateUp()),
+  _onCopyHere: () => {
+    dispatch(FsGen.createCopy())
+    dispatch(navigateUp())
+  },
+  _onMoveHere: () => {
+    dispatch(FsGen.createMove())
+    dispatch(navigateUp())
+  },
+  _onNewFolder: (parentPath: Types.Path) => dispatch(FsGen.createNewFolderRow({parentPath})),
+})
+
+const destinationParentPathIsWritable = memoize(
+  stateProps =>
+    Types.getPathLevel(stateProps._moveOrCopy.destinationParentPath) > 2 &&
+    stateProps._pathItems.get(stateProps._moveOrCopy.destinationParentPath, Constants.unknownPathItem)
+      .writable
+)
+
+const destinationAndSourceAreInSameTLF = memoize(stateProps => {
+  if (!stateProps._moveOrCopy.sourceItemPath) {
+    return false
+  }
+  const src = Types.getPathElements(stateProps._moveOrCopy.sourceItemPath)
+  const dst = Types.getPathElements(stateProps._moveOrCopy.destinationParentPath)
+  return src.length >= 3 && dst.length >= 3 && src[1] === dst[1] && src[2] === dst[2]
```

Comment:
surprised there isn't a Constants helper for this that'll give you the tlf and you can just compare it (even as a string)